### PR TITLE
framework(engine): operation-friendly fail path

### LINKED ISSUE #6806: When user submit a ill-formed DM job, should display the error
### What did you do?

job with wrong format

```
3m3e-server-executor-1               | [2022/08/19 06:05:59.522 +00:00] [ERROR] [server.go:205] ["Failed to create worker"] [error="[code=20023:class=config:scope=internal:level=medium], Message: mysql-instance(0)'s route-rules global not exist in routes, Workaround: Please check the `route-rules` config in task configuration file."] [errorVerbose="[code=20023:class=config:scope=internal:level=medium], Message: mysql-instance(0)'s route-rules global not exist in routes, Workaround: Please check the `route-rules` config in task configuration file.\ngithub.com/pingcap/tiflow/dm/pkg/terror.(*Error).Generate\n\tgithub.com/pingcap/tiflow/dm/pkg/terror/terror.go:271\ngithub.com/pingcap/tiflow/dm/config.(*TaskConfig).adjust\n\tgithub.com/pingcap/tiflow/dm/config/task.go:700\ngithub.com/pingcap/tiflow/dm/config.(*TaskConfig).Adjust\n\tgithub.com/pingcap/tiflow/dm/config/task.go:579\ngithub.com/pingcap/tiflow/engine/jobmaster/dm/config.(*JobCfg).adjust\n\tgithub.com/pingcap/tiflow/engine/jobmaster/dm/config/config.go:229\ngithub.com/pingcap/tiflow/engine/jobmaster/dm/config.(*JobCfg).Decode\n\tgithub.com/pingcap/tiflow/engine/jobmaster/dm/config/config.go:141\ngithub.com/pingcap/tiflow/engine/jobmaster/dm.dmJobMasterFactory.DeserializeConfig\n\tgithub.com/pingcap/tiflow/engine/jobmaster/dm/dm_jobmaster.go:77\ngithub.com/pingcap/tiflow/engine/framework/registry.(*registryImpl).CreateWorker\n\tgithub.com/pingcap/tiflow/engine/framework/registry/registry.go:94\ngithub.com/pingcap/tiflow/engine/executor.(*Server).makeTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:196\ngithub.com/pingcap/tiflow/engine/executor.(*Server).PreDispatchTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:222\ngithub.com/pingcap/tiflow/engine/enginepb._Executor_PreDispatchTask_Handler\n\tgithub.com/pingcap/tiflow/engine/enginepb/executor.pb.go:446\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\tgoogle.golang.org/grpc@v1.46.2/server.go:1283\ngoogle.golang.org/grpc.(*Server).handleStream\n\tgoogle.golang.org/grpc@v1.46.2/server.go:1620\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\tgoogle.golang.org/grpc@v1.46.2/server.go:922\nruntime.goexit\n\truntime/asm_amd64.s:1594\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/engine/framework/registry.(*registryImpl).CreateWorker\n\tgithub.com/pingcap/tiflow/engine/framework/registry/registry.go:96\ngithub.com/pingcap/tiflow/engine/executor.(*Server).makeTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:196\ngithub.com/pingcap/tiflow/engine/executor.(*Server).PreDispatchTask\n\tgithub.com/pingcap/tiflow/engine/executor/server.go:222\ngithub.com/pingcap/tiflow/engine/enginepb._Executor_PreDispatchTask_Handler\n\tgithub.com/pingcap/tiflow/engine/enginepb/executor.pb.go:446\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\tgoogle.golang.org/grpc@v1.46.2/server.go:1283\ngoogle.golang.org/grpc.(*Server).handleStream\n\tgoogle.golang.org/grpc@v1.46.2/server.go:1620\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\tgoogle.golang.org/grpc@v1.46.2/server.go:922\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
```

### What did you expect to see?

precheck should report this error to user

### What did you see instead?

```
$ curl http://127.0.0.1:10245/api/v1/jobs/385bacfa-0aca-40ec-bf83-75f3f5bf0105/status
{
    "error": "job is not online"
}%  
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

```console
(paste Dataflow Engine version here)
```

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #6806

#### Tests 

 - Integration test

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

File: engine/test/integration_tests/dm_full_mode/run.sh [hunk 1]
```diff
@@ -14,6 +14,18 @@ function run() {
...
+	# test a ill-formatted job should fail
+
+	cp $CUR_DIR/conf/job.yaml $WORK_DIR/job.yaml
+  sed -i "20,23d" $WORK_DIR/job.yaml
+  create_job_json=$(base64 -w0 $WORK_DIR/job.yaml | jq -Rs '{ type: "DM", config: . }')
+  echo "create_job_json: $create_job_json"
+  job_id=$(curl -X POST -H "Content-Type: application/json" -d "$create_job_json" "http://127.0.0.1:10245/api/v1/jobs?tenant_id=dm_full_mode&project_id=dm_full_mode" | jq -r .id)
+  echo "job_id: $job_id"
+
+  read -p 123
```
[the remark this answers is not in the document]
Comment (amyangfei) [reply engine/test/integration_tests/dm_full_mode/run.sh:26]: This is a known bug, will be fixed in https://github.com/pingcap/tiflow/pull/7061

[the remark this answers is not in the document]
Comment (amyangfei) [reply engine/test/integration_tests/dm_full_mode/run.sh:26]: After fixed, sample output

```json
{
  "id": "229c3958-4425-4cb8-baf2-1fae49030ef6",
  "type": "DM",
  "state": "Failed",
  "config": "dGFzay1tb2RlOiBmdWxsCnRhcmdldC1kYXRhYmFzZToKICBob3N0OiAxMjcuMC4wLjEKICBwb3J0OiA0MDAwCiAgdXNlcjogcm9vdAogIHBhc3N3b3JkOiAnJwp1cHN0cmVhbXM6CiAgLSBkYi1jb25maWc6CiAgICAgIGhvc3Q6IDEyNy4wLjAuMQogICAgICBwb3J0OiAzMzA2CiAgICAgIHVzZXI6IHJvb3QKICAgICAgcGFzc3dvcmQ6ICcxMjM0NTYnCiAgICBzb3VyY2UtaWQ6IG15c3FsLTAxCiAgICBibG9jay1hbGxvdy1saXN0OiBiYWxpc3QtMDEKICAgIHJvdXRlLXJ1bGVzOiBbICJnbG9iYWwiIF0KYmxvY2stYWxsb3ctbGlzdDoKICBiYWxpc3QtMDE6CiAgICBkby1kYnM6CiAgICAgIC0gZG1fZnVsbCoKIyByb3V0ZXM6CiMgICBnbG9iYWw6CiMgICAgIHNjaGVtYS1wYXR0ZXJuOiAiZG1fZnVsbF9yb3V0ZV9zY2hlbWEiCiMgICAgIHRhcmdldC1zY2hlbWE6ICJkbV9mdWxsX3JvdXRlX3NjaGVtYV9zdWNjIgo=",
  "detail": "",
  "error": {
    "code": "None",
    "message": "[code=20023:class=config:scope=internal:level=medium], Message: mysql-instance(0)'s route-rules global not exist in routes, Workaround: Please check the `route-rules` config in task configuration file.",
    "not_leader": null
  },
  "selectors": []
}
```

File: engine/executor/server.go [hunk 2]
```diff
@@ -268,6 +268,15 @@ func precheckMasterMeta(
...
 ) error {
+	if pkgClient.ErrCreateWorkerTerminate.Is(err) {
+		return rpcerror.ToGRPCError(err)
+	}
+
+	switch tp {
+	case frameModel.DMJobMaster:
+		err = errors.ToDMError(err)
+	default:
+	}
```
Comment (amyangfei) [inline engine/executor/server.go:279]: Can remove this code snippet, since it is contained in `checkBusinessErrorIsRetryable` at L 280.

**Review by amyangfei — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Comment by lance6716** [comment]
```
[2022-09-26T09:22:28.918Z] retry 200 time(s), cmd:  curl "http://127.0.0.1:10245/api/v1/jobs/97e3db63-1e7e-4ea6-aceb-398347838ed8/status" | tee /dev/stderr | jq -e '.task_status."mysql-01".status.status | .finishedBytes > 0 and .finishedBytes < .totalBytes'

[2022-09-26T09:22:28.918Z]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current

[2022-09-26T09:22:28.918Z]                                  Dload  Upload   Total   Spent    Left  Speed

[2022-09-26T09:22:29.175Z] 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
{

[2022-09-26T09:22:29.175Z]     "job_id": "97e3db63-1e7e-4ea6-aceb-398347838ed8",

[2022-09-26T09:22:29.175Z]     "task_status": {

[2022-09-26T09:22:29.175Z]         "mysql-01": {

[2022-09-26T09:22:29.175Z]             "expected_stage": "Running",

[2022-09-26T09:22:29.175Z]             "worker_id": "2d984ffe-b501-490b-b317-d069023b64bb",

[2022-09-26T09:22:29.175Z]             "config_outdated": false,

[2022-09-26T09:22:29.175Z]             "status": {

[2022-09-26T09:22:29.175Z]                 "error_message": "",

[2022-09-26T09:22:29.175Z]                 "unit": "DMLoadTask",

[2022-09-26T09:22:29.175Z]                 "stage": "Running",

[2022-09-26T09:22:29.175Z]                 "result": null,

[2022-09-26T09:22:29.175Z]                 "status": {

[2022-09-26T09:22:29.175Z]                     "finishedBytes": "0",

[2022-09-26T09:22:29.175Z]                     "metaBinlog": "(ON.000003, 503631)",

[2022-09-26T09:22:29.175Z]                     "metaBinlogGTID": "f1f2c113-3d7a-11ed-92d0-0242ac120006:1-2007",

[2022-09-26T09:22:29.175Z]                     "progress": "0.00 %",

[2022-09-26T09:22:29.175Z]                     "totalBytes": "0"
```

🤔

**Comment by lance6716** [comment]
Downstream TiDB is very slow on creating 500 tables. Will increase wait time

**Pull request merged**